deploy/examples/cr-istio.yaml

```diff
@@ -209,3 +209,99 @@ spec:
   resources:
     requests:
       storage: 1Gi
+
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: vault
+  namespace: vault
+
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vault-secrets
+  namespace: vault
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    verbs:
+      - "*"
+
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vault-secrets
+  namespace: vault
+roleRef:
+  kind: Role
+  name: vault-secrets
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: vault
+
+---
+# This binding allows the deployed Vault instance to authenticate clients
+# through Kubernetes ServiceAccounts (if configured so).
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: vault-auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+  - kind: ServiceAccount
+    name: vault
+    namespace: vault
+
```

Comment:
I think you could leave this out as it is not much different from deploy/default/rbac.yaml